test/jdk/java/text/Format/DecimalFormat/MantissaDigits.java

```diff
@@ -0,0 +1,116 @@
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8159023
+ * @summary Confirm behavior of mantissa for scientific notation in Decimal Format
+ * @run junit MantissaDigits
+ */
+
+import java.text.DecimalFormat;
+import java.util.Locale;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class MantissaDigits {
+    private static final double[] numbers = {
+            1.1, 12.1, 123.1, 1234.1, 12345.1, 123456.1,
+            -1.1, -12.1, -123.1, -1234.1, -12345.1, -123456.1,
+            1, 12, 123, 1234, 12345, 123456, 1234567,
+            -1, -12, -123, -1234, -12345, -123456, -1234567,
+            1.1234, 1.1111, 1.412, 222.333, -771.2222
+            };
+
+    @ParameterizedTest
+    @MethodSource("patterns")
+    public void testMantissaDefinition(String pattern) {
+        Locale.setDefault(Locale.ENGLISH);
+        DecimalFormat df = new DecimalFormat(pattern);
+        for (double number : numbers) {
+            // Count the significant digits in the pre-formatted number
+            int originalNumDigits = String.valueOf(number)
+                    .replace(".", "")
+                    .replace("-", "")
+                    .length();
+
+            if (wholeNumber(number)) {
+                // Trailing 0 should not be counted
+                originalNumDigits--;
+            }
+
+            // Format the number, then grab the significant
+            // digits inside the mantissa
+            String formattedNum = df.format(number);
+            int mantissaDigits = formattedNum
+                    .replace(".", "")
+                    .replace("E", "")
+                    .replace("-", "")
+                    .length() - 1;
+
+            // Count the zeroes and hash symbols in the
+            // pattern, to be used in the mathematical expression
+            int zeroes = (int) (pattern.chars()
+                    // ignore exponent zero
+                    .filter(ch -> ch == '0').count() - 1);
+            int hashes = (int) pattern.chars()
+                    .filter(ch -> ch == '#').count();
+
+
+            // Test the new definition of the Mantissa
+            Integer calculatedDigits = Math
+                    .min(Math.max(zeroes, originalNumDigits), (hashes+zeroes));
```

Comment:
Could we make those expected numbers into the `Arguments` so that there is no need to calculate in the test case at runtime? I think that would be more readable.

Reply:
Thanks for the review, addressed both suggestions in the most recent commit.